packages/dev/core/src/assetContainer.ts

```diff
@@ -376,7 +481,7 @@ export class AssetContainer extends AbstractScene {
             if (predicate && !predicate(o)) {
                 return;
             }
-            this.scene.addMesh(o);
+            this.scene.addMesh(o, true);

```

Comment:
Was it a bug in the previous code?

I would have thought that `this.meshes` contains all the meshes, so there's no need to recursively add, all meshes will eventually be added to the scene?

Moreover, by using the recursive flag, all descendants of `o` won't be checked by the predicate.

Reply:
I was a bit unsure about what should the expected behavior here, because the user could manually add just some of the meshes in the hierarchy to the container, and in this case currently addAll/removeAll will only affect the manually added meshes: https://playground.babylonjs.com/#5NFRVE#160 while instantiateAll will add all meshes.

Reply:
Fair, I can remove the true and create an issue for tracking?

Reply:
Created issue: https://github.com/BabylonJS/Babylon.js/issues/13477